### ISSUE TITLE
DirectionsRoute.geometry NullPointerException

### ISSUE THREAD

**Issue by Guardiola31337** [mapbox/mapbox-navigation-android#2716]
Run into the following NPE when testing the `SimpleMapboxNavigationKt` example from `examples` test app 👀 

```
04-06 18:36:13.753 24058-25088/? E/AndroidRuntime: FATAL EXCEPTION: Thread-15828
    Process: com.mapbox.navigation.examples, PID: 24058
    java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.mapbox.api.directions.v5.models.DirectionsRoute.geometry()' on a null object reference
        at com.mapbox.navigation.ui.route.FeatureProcessingTask.createRouteFeatureCollection(FeatureProcessingTask.java:77)
        at com.mapbox.navigation.ui.route.FeatureProcessingTask.run(FeatureProcessingTask.java:47)
```

cc @abhishek1508

**Comment by Guardiola31337** [comment]
Steps to reproduce 👀 

![feature_processing_task_npe](https://user-images.githubusercontent.com/1668582/78828886-66dba900-79b3-11ea-87e0-db1b08f13335.gif)

Refs. https://github.com/mapbox/mapbox-navigation-android/issues/2654

**Comment by abhishek1508** [comment]
This is caused as a result of regression from this [PR](https://github.com/mapbox/mapbox-navigation-android/pull/2703)

**Comment by kmadsen** [comment]
@abhishek1508 the stack trace looks like it's happening in `FeatureProcessingTask`

https://github.com/mapbox/mapbox-navigation-android/pull/2703 was fixing a NullPointerException caused by the same property, possibly stealing the earliest crash (otherwise we should be able to fix the issue in MapRouteProgressChangeListenerTest)

This looks like an issue that needs to be resolved in FeatureProcessingTask though

**Comment by kmadsen** [comment]
Kotlin is giving the warning here

![Screen Shot 2020-04-10 at 4 16 20 PM](https://user-images.githubusercontent.com/3021882/79028849-a1297f80-7b46-11ea-91fa-a9498427a47a.png)

**Comment by kmadsen** [comment]
Yep you're right @abhishek1508. Here it is https://github.com/mapbox/mapbox-navigation-android/pull/2754

**Comment by kmadsen** [comment]
The fix does feel a little weird https://github.com/mapbox/mapbox-navigation-android/pull/2754

Because the `MapRouteArrow` can handle null geometry. This issue may be in `MapRouteLine`

**Comment by kmadsen** [comment]
@abhishek1508 https://github.com/mapbox/mapbox-navigation-android/pull/2703/files#diff-a8a708e2a20d5c4ba94cf7fdcc47b618L31

The old code did not check for `null` `geometry` so this must have been introduced by Free Drive things like you mentioned.

This will fix it #2754, and it's covered with tests. So I'm happy with it. But we may see it come up again :)
